tests/conftest.py

```diff
@@ -19,19 +21,29 @@ class D(Base):
     class C(D, E):
         friend: str
 
-    class B(Base):
-        my: str
-        old: list[C]
-        right: "str | None"
-
     class G(Base):
         g: str
 
     class H(Base):
         h: str
 
-    class A(Base):
-        hello: str
-        darkness: B
+    if sys.version_info >= (3, 10):
+        class B(Base):
+            my: str
+            old: list[C]
+            right: "str | None"
+
+        class A(Base):
+            hello: str
+            darkness: B
+    else:
+        class B(Base):
+            my: str
+            old: List[C]
+            right: Optional["str"]
+
+        class A(Base):
+            hello: str
+            darkness: B
```

Comment:
This portion should be included both for 3.8 and higher than 3.8 because we need to guarantee that imports from typing are going to continue working in higher python versions.

Reply:
My idea was to keep tests effectively the same for Python >= 3.10 in this PR, and introduce equivalent tests for lower versions. I totally agree imports from typing should be properly tested for all versions, but this was not the goal of this issue, and I think it should be done differently than via fields in A and B schemas. I'd like to keep the scope of this PR to the 3.8/3.9 support. How about addressing types testing in #10?